File: 162. Find Peak Element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        if (nums.size() == 1 || nums[0] > nums[1])
            return 0;
        else if (nums[nums.size() - 2] < nums[nums.size() - 1])
            return nums.size() - 1;
        else
        {
            for (int i = 1; i < nums.size() - 1; i++)
            {
                if (nums[i] > nums[i-1] && nums[i] > nums[i+1])
                    return i;
            }
        }
        return 0;
    }
};

File: 162. Find Peak Element.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 1 or nums[0] > nums[1]:
            return 0
        elif nums[len(nums)-2] < nums[len(nums)-1]:
            return len(nums)-1
        else:
            for i in xrange(1, len(nums)-1):
                if nums[i] > nums[i-1] and nums[i] > nums[i+1]:
                    return i
